pkg/armadaevents/events.proto

```diff
@@ -290,6 +291,17 @@ message JobRunLeased {
     string node_id = 4;
     // Used by the scheduler to maintain a consistent state
     int32 update_sequence_number = 5;
+    // This bool is here because it is the most ergonomic way of distinguishing
+    // the case where `scheduled_at_priority` was explicitly set to zero from
+    // the case where it was not set at all.
```

Comment:
The alternatives are:

1. Use `Int32Value` instead of `int32` for `scheduled_at_priority`; this has the downside that switching between these types is a breaking change.
2. Wrap `scheduled_at_priority` in `oneof`; this has the downside that the generated code is ugly (and no safer than this approach; for example, `github.com/gogo/protobuf` still generates a `GetScheduledAtPriority` getter that returns an `int32`).
3. Use the `optional` keyword; this is not supported by `github.com/gogo/protobuf`.

All of these seem worse to me.